core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/CredentialParametersValidator.java

```diff
@@ -67,4 +73,85 @@ private boolean validateParams(Map<String, Object> params, ConstraintValidatorCo
         return true;
     }
 
+    private boolean validateOpenStackParams(Map<String, Object> params, ConstraintValidatorContext context, List<TemplateParam> paramConstraints) {
+        boolean valid = true;
+        String keystoneVersion = String.valueOf(params.get(OpenStackCredentialParam.KEYSTONE_VERSION.getName()));
+        if (keystoneVersion.equals(KEYSTONE_VERSION_V2)) {
```

Comment:
if no keystone version provided, handle as v2 for backward compatibilty:
if (keystoneVersion.equals("null") || keystoneVersion.equals(KEYSTONE_VERSION_V2)) {